lib/Sema/CSGen.cpp

```diff
@@ -1115,11 +1115,25 @@ namespace {
       auto baseTy = base->getType();
       auto fnTy = FunctionType::get(inputTv, outputTy);
 
+      // FIXME: This can only happen when diagnostics successfully type-checked
+      // sub-expression of the subscript and mutated AST, but under normal
+      // circumstances subscript should never have InOutExpr as a direct child.
```

Comment:
Slight correction: It's not that subscript shouldn't have InOutExpr as a child, it's that it shouldn't happen in un-typechecked ASTs.